mount.go

```diff
@@ -16,5 +16,78 @@
 
 package virtcontainers
 
+import (
+	"fmt"
+	"path/filepath"
+	"syscall"
+)
+
 // These mounts need to be created by the agent within the VM
 var systemMounts = []string{"/proc", "/dev", "/dev/pts", "/dev/shm", "/dev/mqueue", "/sys", "/sys/fs/cgroup"}
+
+func major(dev uint64) int {
+	return int((dev >> 8) & 0xfff)
+
+}
+
+func minor(dev uint64) int {
+	return int((dev & 0xff) | ((dev >> 12) & 0xfff00))
+}
+
+type device struct {
+	major      int
+	minor      int
+	mountPoint string
+}
+
+// getDeviceForPath gets the underlying device containing the file specified by path.
+// The device type constitutes the major-minor number of the device and the dest mountPoint for the device
+func getDeviceForPath(path string) (*device, error) {
+	if path == "" {
+		return &device{}, fmt.Errorf("Path cannot be empty")
+	}
+
+	stat := syscall.Stat_t{}
+	err := syscall.Stat(path, &stat)
+	if err != nil {
+		return &device{}, err
+	}
+
+	// stat.Dev points to the underlying device containing the file
+	major := major(stat.Dev)
+	minor := minor(stat.Dev)
+
+	mountPoint := path
+
+	// We get the mount point by recursively peforming stat on the path
+	// The point where the device changes indicates the mountpoint
+	for {
+		if mountPoint == "/" {
```

Comment:
@amshinde let us take two cases.

If you have /mydisk where mydisk is just a directory on / 
vs /mydisk where you have mounted something on /mydisk

In the first case /mydisk and / will have the same stat
In the second case /mydisk and / will have different stats.

So in the first case we error out saying we did not find a mount point.
In the second case /mydisk is the mount point.

The only time / is a valid mount point will be in the case of volumes where the user may have done something along the lines of -v /:/hostdisk.
In such a case yes / is the mount point. But that is a special case, where the source is /.